### ISSUE TITLE
[Feature Request] Ability to provide a custom count function for the counter in the v-text-field

### ISSUE THREAD

**Issue by Shinigami92** [vuetifyjs/vuetify#8788]
### Problem to solve
We have some `v-text-field` that holds text with variables like `this is my text with a ${variable}`.
Then we have multiple values that can placed into this variable and we choose the top length. So there is in example a value `hello world example`. It is 19 characters long + the 23 characters from the input text. So now we allow only a maximum of 30 characters, so the counter should show `42 / 30` in red.

### Proposed solution
This could be solved with a new optional `prop` that takes the current `input` of the `v-text-field` and returns a `number`. We could name it `counter-length` or `length` or `input-length` or something similar.

So we use `<v-text-field v-model="line" :counter="30" :counter-length="myCustomCounter">`
```ts
myCustomCounter(value: string): number {
  // implementation
}
```

---

It would also be nice if the `:counter-length` could be an `async function`, because we calculate the length via backend / api.

**Comment by myleslee** [comment]
There's another user case that validates this request. In one of our projects we need to deal with a legacy database that doesn't support unicode. So a `varchar(15)` column can hold up to 15 latin letters, or 7 Chinese characters plus 1 latin letter (7*2 + 1 = 15). 

The current built-in Counter of `VTextField` will report 7 Chinese characters plus 1 latin letter as 8,  which would result in lengthy data to be rejected by the database.

So it would be awesome if we could provide our own counter logic to the `VTextField`.

**Comment by Shinigami92** [comment]
Should I do anything else in my pull request to please someone?

**Comment by Shinigami92** [comment]
closed by #8802
